appserver/web/web-core/src/main/java/org/apache/catalina/core/StandardContext.java

```diff
@@ -4977,37 +4977,58 @@ public boolean alternateResourcesStop() {
      */
     public void loadOnStartup(Container children[]) throws LifecycleException {
         // Collect "load on startup" servlets that need to be initialized
-        Map<Integer, List<Wrapper>> map = new TreeMap<>();
+        Map<Integer, List<Wrapper>> loadOnStartupServlets = new TreeMap<>();
+        List<Wrapper> nonLoadOnStartupServlets = new ArrayList<>();
+
         for (Container aChildren : children) {
             Wrapper wrapper = (Wrapper) aChildren;
+
             int loadOnStartup = wrapper.getLoadOnStartup();
             if (loadOnStartup < 0) {
-                continue;
+                nonLoadOnStartupServlets.add(wrapper);
+            } else {
+                loadOnStartupServlets.computeIfAbsent(loadOnStartup, e -> new ArrayList<>())
```

Comment:
```suggestion
                loadOnStartupServlets.computeIfAbsent(loadOnStartup, ArrayList::new)
```

Reply:
I think arraylist::new will not work, as the key will be input into the ctor. With the key being an integer and the ctor setting the size, they will cause an out of memory with the huge key values being used.

Reply:
Wow, Arjan is right, I had the same idea, we discussed it in a chat and I started googling:
https://community.sonarsource.com/t/java-bug-using-collectors-tomap-with-arraylist-new/7661

Reply:
I'm happy to learn it now and not from the GF bug :smile: